### PR TITLE
Group email alias creation on group-create event

### PR DESCRIPTION
**Pull request by stephdl**
https://github.com/NethServer/dev/issues/5581

### CODE REVIEW

File: root/usr/share/nethesis/NethServer/Language/en/NethServer_Module_Group.php [hunk 1]
```diff
@@ -14,3 +14,5 @@
 $L['groupname_label'] = 'Group name'; 
 $L['valid_platform,group-create,group-length,1'] = 'Name is too long: ${reason}';
 $L['group_exists'] = 'The group already exists';
+$L['CreatePseudoRecords_label'] = 'Create also a shared mailbox and an email address named after the group';
```
Comment (DavidePrincipi) [inline root/usr/share/nethesis/NethServer/Language/en/NethServer_Module_Group.php]: What about removing "also"?

    Create a shared mailbox and an email address named after the group

**Comment by DavidePrincipi** [comment]
Could you attach also a screenshot of the resulting UI?

**Comment by stephdl** [comment]
screenshot_2018-09-17 example org - utilisateurs et groupes

File: root/usr/share/nethesis/NethServer/Module/Account/Group/Modify.php [hunk 2]
```diff
@@ -158,6 +161,30 @@ public function process()
...
+            if ($this->parameters['CreatePseudoRecords'] !== 'yes') {
+                return;
+            }
+            $members = implode(",",array_unique($this->parameters['members']));
+            $accountsDb = $this->getPlatform()->getDatabase('accounts');
+            $domain = $this->getPlatform()->getDatabase('configuration')->getType('DomainName');
+            $groupnameFull = $this->parameters['groupname'] . '@' . $domain;
+
+            if( ! $accountsDb->getKey($groupnameFull)) {
+                $accountsDb->setKey($groupnameFull, 'pseudonym', array ('Description'=>'Group Alias for postfix', 'Account' => $members));
```
[the remark this answers is not in the document]
Comment (stephdl) [reply root/usr/share/nethesis/NethServer/Module/Account/Group/Modify.php]: ok to show it

**Comment by stephdl** [comment]
@DavidePrincipi, after thinking, there is no interest for the Description field, it is not part of the group properties, it is part of a key with a pseudonym type.

**Comment by stephdl** [comment]
I think that ur patch is not cmplete, we should make possible to revert it, and add/remove users in the group alias...what do you think ?

File: root/usr/share/nethesis/NethServer/Module/Account/Group/Modify.php [hunk 3]
```diff
@@ -158,6 +161,29 @@ public function process()
...
+
+            $accountsDb = $this->getPlatform()->getDatabase('accounts');
+            $domain = $this->getPlatform()->getDatabase('configuration')->getType('DomainName');
+            $groupnameFull = $this->parameters['groupname'] . '@' . $domain;
+            $members= array_unique($this->parameters['members']);
+            $members[] = $groupnameFull;
+            $members = implode(",",$members);
+
+            if( ! $accountsDb->getKey($groupnameFull)) {
+                $accountsDb->setKey($groupnameFull, 'pseudonym', array ('Description'=>'Automatic group mailbox', 'Account' => $members));
```
Comment (DavidePrincipi) [inline root/usr/share/nethesis/NethServer/Module/Account/Group/Modify.php]: TL;DR remove `$members` variable, set "Account" to `vmail+$this->parameters['groupname']`

----

Well, I checked again our code [here](https://github.com/NethServer/nethserver-mail/blob/master/server/usr/share/nethesis/NethServer/Module/MailAccount/Pseudonym/Modify.php#L173) and I realized I put you in the wrong direction.

There is just one destination/target "Account"  for our pseudonym. When the destination is a shared mailbox, its syntax uses an extension address, like `vmail+mbxname`, without domain part.

To clarify, this is an example:

```text
repomaster@domain.tld=pseudonym
    Access=public
    Account=vmail+repomaster
    ControlledBy=system
    Description=
```

**Comment by DavidePrincipi** [comment]
Please review my code change proposal: https://github.com/stephdl/nethserver-sssd/pull/1

**Pull request closed without merging**